code/go/pkg/validator/validator_fields_test.go

```diff
@@ -0,0 +1,521 @@
+			},
+		},
+		{
+			title:           "runtime fields: wrong type",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":    "runtime_field",
+					"type":    "byte",
+					"runtime": true,
+				},
+			},
+			expectedErrors: []string{
+				`field 0: Must not be present`,
+			},
+		},
+		{
+			title:           "runtime fields: invalid runtime value",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":    "runtime_field",
+					"type":    "boolean",
+					"runtime": 30,
+				},
+			},
+			expectedErrors: []string{
+				`field 0.runtime: Invalid type. Expected: string, given: integer`,
+			},
+		},
+		{
+			title:           "runtime fields: invalid runtime boolean",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":    "runtime_field",
+					"type":    "binary",
+					"runtime": true,
+				},
+			},
+			expectedErrors: []string{
+				`field 0: Must not be present`,
+			},
+		},
+		{
+
+			title:           "runtime fields: invalid runtime boolean",
+			packageTemplate: "integration_v3_0",
+			fields: []map[string]any{
+				{
+					"name":    "runtime_field",
+					"type":    "binary",
+					"runtime": "doc['message'].value().doSomething()",
+				},
+			},
+			expectedErrors: []string{
+				`field 0: Must not be present`,
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.title, func(t *testing.T) {
+			packagePath := createPackageWithFields(t, "testpackage", c.packageTemplate, c.fields)
+			err := ValidateFromPath(packagePath)
+			if len(c.expectedErrors) == 0 {
+				assert.NoError(t, err)
+				return
+			}
+			require.Error(t, err)
+
+			errs, ok := err.(specerrors.ValidationErrors)
+			require.True(t, ok)
+			assert.Len(t, errs, len(c.expectedErrors))
+
+			for _, foundError := range errs {
+				// Trim the part of the error that refers to the file where the error was found.
+				_, trimmedError, found := strings.Cut(foundError.Error(), ": ")
+				if assert.True(t, found) {
+					assert.Contains(t, c.expectedErrors, trimmedError)
+				}
+			}
+		})
+	}
+}
+
+func createPackageWithFields(t *testing.T, pkgName string, pkgTemplate string, fields any) string {
+	var encodedFields []byte
+	switch fields.(type) {
+	case nil:
+	case []byte:
+		encodedFields = fields.([]byte)
+	default:
+		var jsonFields bytes.Buffer
+		enc := yaml.NewEncoder(&jsonFields)
+		err := enc.Encode(fields)
+		require.NoError(t, err)
+		encodedFields = jsonFields.Bytes()
+	}
+
+	return createPackageWithRawFields(t, pkgName, pkgTemplate, encodedFields)
+}
+
+func createPackageWithRawFields(t *testing.T, pkgName string, pkgTemplate string, rawFields []byte) string {
+	basePackageDir := filepath.Join("testdata", "templates", pkgTemplate)
+	require.DirExists(t, basePackageDir, "checking if template package exists")
+
+	pkgDir := filepath.Join(t.TempDir(), pkgName)
+	err := cp.Copy(basePackageDir, pkgDir)
+	require.NoError(t, err, "copying template package")
```

Comment:
One option to set spec version dynamically could be setting a specific string for the `format_version` field in the manifest file for each template. For instance:
```
format_version: <to_be_set>
```

This could be replaced by tests when creating the package with something like:

```suggestion
    basePackageDir := filepath.Join("testdata", "templates", pkgTemplate)
    require.DirExists(t, basePackageDir, "checking if template package exists")

    packageManifest := filepath.Join(basePackageDir, "manifest.yml")
    manifest, err := os.ReadFile(packageManifest)
    require.NoError(t, err)

    output := bytes.ReplaceAll(manifest, []byte("format_version: <to_be_set>"), []byte(fmt.Sprintf("format_version: %s", pkgVersion)))

    err = os.WriteFile(packageManifest, output, 0644)
    require.NoError(t, err)

    pkgDir := filepath.Join(t.TempDir(), pkgName)
    err = cp.Copy(basePackageDir, pkgDir)
    require.NoError(t, err, "copying template package")
```

Reply:
I would leave a default version and then optionally allow to override it. I will give it a try.